### PR TITLE
Refactor product page featured products section

### LINKED ISSUE #1206: Featured Products on ProductDetailPage has no height limit
Featured Products on ProductDetailPage has no height limit so the section become rather tall to accommodate the square five-across product grid.  And not all product images are sharp at this size (might be a separate issue/problem). 

image

https://www.ifixit.com/products/fixmat

I'd propose we set a max height for this section and pull in more products to the section. It might make sense to convert this section to a carousel while we're at it.

### LINKED ISSUE #1434: Use new product grid component in product template featured products section

### PR DESCRIPTION
**Pull request by dhmacs**
closes #1434 
closes #1206 

## QA

1. Open [product page preview](https://react-commerce-git-refactor-product-page-featured-dc6043-ifixit.vercel.app/products/repair-business-toolkit)
2. Verify that the featured products section adheres to [mockup](https://www.figma.com/file/jRG4EyQXRDMGfyLqut08KP/iFixit---Working-files?node-id=6650-80646&t=lm5sZ8asm7PGe46K-4) (Note: The title is the same of production, and the description in the default scenario is not shown. This is something that will be configurable when the section config will come from Strapi)

    Screenshot 2023-03-08 at 17 31 51

   > 👁️ Note that on wider screens the section displays 6 columns to satisfy #1206 requirements

3. Verify that the section is responsive

### CODE REVIEW

**Comment by deltuh-vee** [comment]
QA 🎬
The Featured Products section is responsive

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻

File: frontend/lib/strapi-sdk/generated/sdk.ts [hunk 1]
```diff
@@ -2667,118 +2667,6 @@ export type FindStoreQuery = {
                      } | null;
                   } | null;
                } | null;
-               menu3?: {
```
Comment (danielbeardsley) [inline frontend/lib/strapi-sdk/generated/sdk.ts:2670]: This pull didn't intend to remove this I don't think. This was added in #1444 and I think accidentally removed here. @aburke07

Comment (aburke07) [reply frontend/lib/strapi-sdk/generated/sdk.ts:2670]: Ah! Thanks for finding this Danny!

Comment (masonmcelvain) [reply frontend/lib/strapi-sdk/generated/sdk.ts:2670]: It caused a diff every time you ran a dev server, so I thought removing it was nice for DX

Comment (dhmacs) [reply frontend/lib/strapi-sdk/generated/sdk.ts:2670]: This is autogenerated, so I wonder how it is possible that this was removed 🤔

Comment (masonmcelvain) [reply frontend/lib/strapi-sdk/generated/sdk.ts:2670]: I think there was some confusion, this pull has more context: https://github.com/iFixit/react-commerce/pull/1455

Comment (masonmcelvain) [reply frontend/lib/strapi-sdk/generated/sdk.ts:2670]: TLDR, nothing needs to change, menu3 shouldn't have been in the sdk in the first place

**Pull request merged**